ksp_plugin_test/plugin_integration_test.cpp

```diff
@@ -0,0 +1,634 @@
+    Vector<double, AliceSun> const unit_tangent = Normalize(tangent);
+    EXPECT_THAT(
+        InnerProduct(unit_tangent,
+                     satellite_initial_displacement_ /
+                         satellite_initial_displacement_.Norm()),
+        Eq(0));
+    // This yields a circular orbit.
+    satellite_initial_velocity_ =
+        Sqrt(bodies_[SolarSystem::kEarth]->gravitational_parameter() /
+                 satellite_initial_displacement_.Norm()) * unit_tangent;
+  }
+
+  DegreesOfFreedom<Barycentric> ICRFToBarycentric(
+      DegreesOfFreedom<ICRFJ2000Ecliptic> const& degrees_of_freedom) {
+    return {icrf_to_barycentric_positions_(degrees_of_freedom.position()),
+            ircf_to_barycentric_linear_(degrees_of_freedom.velocity())};
+  }
+
+  void InsertAllSolarSystemBodies() {
+    for (std::size_t index = SolarSystem::kSun;
+         index < bodies_.size();
+         ++index) {
+      std::unique_ptr<Index> parent_index =
+          index == SolarSystem::kSun
+              ? nullptr
+              : std::make_unique<Index>(SolarSystem::parent(index));
+      DegreesOfFreedom<Barycentric> const initial_state =
+          ICRFToBarycentric(solar_system_->trajectories()[index]->
+                            last().degrees_of_freedom());
+    // barf_cast, to be fixed when we have rvalue conversion.
+    plugin_->DirectlyInsertCelestial(
+        index,
+        parent_index.get(),
+        initial_state,
+        std::move(
+            *reinterpret_cast<std::unique_ptr<MassiveBody>*>(
+                &bodies_[index])));
+    }
+  }
+
+  Identity<ICRFJ2000Ecliptic, Barycentric> ircf_to_barycentric_linear_;
+  AffineMap<ICRFJ2000Ecliptic,
+            Barycentric,
+            Length,
+            Identity> icrf_to_barycentric_positions_;
+  Permutation<ICRFJ2000Ecliptic, AliceSun> looking_glass_;
+  not_null<std::unique_ptr<SolarSystem>> solar_system_;
+  Instant initial_time_;
+  GravitationalParameter sun_gravitational_parameter_;
+  Angle planetarium_rotation_;
+
+  not_null<std::unique_ptr<Plugin>> plugin_;
+
+  // These initial conditions will yield a low circular orbit around Earth.
+  Displacement<AliceSun> satellite_initial_displacement_;
+  Velocity<AliceSun> satellite_initial_velocity_;
+
+ private:
+  SolarSystem::Bodies bodies_;
+};
+
+TEST_F(PluginIntegrationTest, AdvanceTimeWithCelestialsOnly) {
+  InsertAllSolarSystemBodies();
+  plugin_->EndInitialization();
+  Time const δt = 0.02 * Second;
+  Angle const planetarium_rotation = 42 * Radian;
+  // We step for long enough that we will find a new segment.
+  Instant t = initial_time_;
+  for (t += δt; t < initial_time_ + 10 * 45 * Minute; t += δt) {
+    plugin_->AdvanceTime(t, planetarium_rotation);
+  }
+  EXPECT_THAT(
+      RelativeError(
+          plugin_->
+              CelestialFromParent(SolarSystem::kEarth).displacement().Norm(),
+          1 * AstronomicalUnit),
+      Lt(0.01));
```

Comment:
Perhaps, but it's hardly the point. We have tests for the accuracy of our integrator elsewhere (and we could add more based on eclipses &c.), this will catch apocalypses.